resources/[freeroam]/OGPcm/Code/client.lua

```diff
@@ -0,0 +1,63 @@
+--[[
+By Atton_Risk for OGP 2019
+https://github.com/AttonRisk
+]]--
+-- Some are removed but can be used as ramps
+rampTable = {
+	--1245,
+	1503,
+	13593,
+	--13641,
+	13645,
+	1632,
+	1633,
+	1634,
+	1655,
+	--2931,
+	3080
+}
+-- Math clusterfuck
+function getPositionInfrontOfElement ( element , meters )
+    if not element or not isElement ( element ) then
+        return false
+    end
+    if not meters then
+       meters = 3
+    end
+    local posX , posY , posZ = getElementPosition ( element )
+    local _ , _ , rotation = getElementRotation ( element )
+    posX = posX - math.sin ( math.rad ( rotation ) ) * meters
+    posY = posY + math.cos ( math.rad ( rotation ) ) * meters
+    return posX , posY , posZ
+end
+
+local oldRamp = 0
+
+function wipeRamp()
+	if isElement(oldRamp) then
+		destroyElement(oldRamp)
+	end
+end
+setTimer(wipeRamp,1000*30,0)
+
+function spawnRamp(cn,rampID)
+
+	if rampID then
+		wipeRamp()
```

Comment:
I think you can get a much better user experience if you change this around a little.
Currently this will wipe the old ramp every 30 seconds after the resource is started, this could be 1 second after the `spawnRamp` command is called.

Try:
```lua
function wipeRamp(ramp)
    if isElement(ramp) then
        destroyElement(ramp)
    end
end

function spawnRamp(cmd, rampId)
    if rampId then
        wipeRamp(oldRamp)
.....
        setTimer(function()
            wipeRamp(oldRamp)
        end,1000*30,1)
    else
        outputChatBox("Use a vaild ramp!",255,0,0) --Spell check your chat box please
    end
end
addCommandHandler("ramp", spawnRamp)
```
Doing it this way means that the user will only have one ramp at a time, but they get to keep each ramp for 30 seconds if they don't create a new one rather than 0-30 seconds if we do a timer from resource start.

Reply:
`.....` represents lines 48 -> 56 in the original file.